resolve/index.go

```diff
@@ -78,6 +78,20 @@ type RuleIndex struct {
 	importMap      map[ImportSpec][]*ruleRecord
 	mrslv          func(r *rule.Rule, pkgRel string) Resolver
 	crossResolvers []CrossResolver
+
+	// embedded indicates whether another rule of the same language embeds this
+	// the rule. Embedded rules should not be indexed.
+	embedded map[label.Label]bool
```

Comment:
Could this be a `map[label.Label]struct{}` to save memory?

Reply:
You mean combining the 3 maps? I think that should work 🤔

Reply:
I just meant that you don't use the tri-state of `bool` + key existence here, so empty structs as values would save memory. But if you can combine the maps, that's of course even better 😅

Reply:
Interesting, TIL!

I'll switch to `struct{}` but leaving it as 3 maps for now. Maybe that could be a followup 🤔